### PR TITLE
[Net] Removed blocking network address resolution in connect_to_host in websockets

### LINKED ISSUE #53265: connect_to_host functions in network classes block on address resolution
### Godot version

3.x, master

### System information

all

### Issue description

the implementation of connect_to_host in the network classes (other than HttpClient) can end up blocking the main engine thread in `getaddrinfo()`

### Steps to reproduce

Setup a laggy DNS resolver and call `connect_to_host` from a network peer, or websocket client.  The engine main thread will block in `getaddrinfo()`.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by jordo**
Master pr for https://github.com/godotengine/godot/pull/53243 (untested)

*Bugsquad edit:* fixes #53265 for websocket clients

### CODE REVIEW

**Comment by jordo** [comment]
fixes https://github.com/godotengine/godot/issues/53265 for websocket clients

File: modules/websocket/wsl_client.h [hunk 1]
```diff
@@ -63,10 +71,15 @@ class WSLClient : public WebSocketClient {
 
 	String _key;
 	String _host;
-	int _port;
-	Array ip_candidates;
+	uint16_t _port;
+	String _path;
+	Array _ip_candidates;
 	Vector<String> _protocols;
+	Vector<String> _original_protocols;
```
Comment (Faless) [inline modules/websocket/wsl_client.h:78]: I don't understand why we need the `_original_protocols`.
It seems there is already the `_protocols` variable that could be used.

File: modules/websocket/wsl_client.cpp [hunk 2]
```diff
@@ -231,6 +195,84 @@ int WSLClient::get_max_packet_size() const {
...
+		if (err != OK) {
+			_status = Status::STATUS_DISCONNECTED;
+			_tcp->disconnect_from_host();
+			_on_error();
+			return;
+		}
+		_connection = _tcp;
+
+		_key = WSLPeer::generate_key();
+		// TODO custom extra headers (allow overriding this too?)
```
Comment (Faless) [inline modules/websocket/wsl_client.cpp:251]: We can remove this TODO since we now support extra headers (was a leftover from the other PR I guess).

**Review by Faless — CHANGES_REQUESTED** [review]
I am deeply sorry for the very long review time, I have been sucked up by the work on the replication API.
This change looks okay, it is a bit of a mess that this has to be handled down in every peer separately, but it's definitely an improvement so I would merge this once the review comments are addressed.
Good work :+1:

**Pull request closed without merging**